### PR TITLE
Add DataPostprocessorTensor

### LINKED ISSUE #4995: Implement DataPostprocessor[Symmetric]Tensor
[This comment](https://github.com/dealii/dealii/pull/4991#issuecomment-326033087) in #4991 refers.

> We don't currently support tensorial output. But I think it would still be nice to have the equivalent of these classes, say DataPostprocessorTensor and DataPostprocessorSymmetricTensor. They would output individual tensor components as scalars, so maybe that's not quite that useful, but it would still make some things easier.

### PR DESCRIPTION
**Pull request by bangerth**
This adds the `DataPostprocessorTensor` class, along with documentation and a testcase. See also the discussion in #4991.

Closes #4995.

### CODE REVIEW

**Comment by davydden** [comment]
impressive pictures! :+1:

**Comment by bangerth** [comment]
Yes, fun, right? :-)

File: include/deal.II/numerics/data_postprocessor.h [hunk 1]
```diff
@@ -816,6 +820,245 @@ class DataPostprocessorVector : public DataPostprocessor<dim>
...
+ * The only tricky part in this piece of code is how to sort the <code>dim*dim</code>
+ * elements of the strain tensor into the one vector of computed output
+ * quantities -- in other words, how to <i>unroll</i> the elements of
+ * the tensor into the vector. This is facilitated by the
+ * Tensor::component_to_unrolled_index() function that takes a
+ * pair of indices that specify a particular element of the
+ * tensor and returns a vector index that is then used in the code
+ * above to fill the @p computed_quantities array.
+ *
+ * The only thing that is necessary is to add another output to the call
```
Comment (jppelteret) [inline include/deal.II/numerics/data_postprocessor.h]: "The last/final thing"?

File: include/deal.II/numerics/data_postprocessor.h [hunk 2]
```diff
@@ -816,6 +820,245 @@ class DataPostprocessorVector : public DataPostprocessor<dim>
...
+ *
+ * Using this class in in step-8 leads to the following visualization:
+ *
+ * @image html data_postprocessor_tensor_2.png
+ *
+ * Given how easy it is to output the strain, it would also not be very
+ * complicated to write a postprocessor that computes the <i>stress</i>
+ * in the solution field as the stress is easily computed from the
+ * strain by multiplication with either the strain-stress tensor or,
+ * in simple cases, the Lame constants.
```
Comment (jppelteret) [inline include/deal.II/numerics/data_postprocessor.h]: "Lamé"

File: include/deal.II/numerics/data_postprocessor.h [hunk 3]
```diff
@@ -816,6 +820,245 @@ class DataPostprocessorVector : public DataPostprocessor<dim>
...
+
+  /**
+   * Return the vector of strings describing the names of the computed
+   * quantities. Given the purpose of this class, this is a vector with dim
+   * entries all equal to the name given to the constructor.
+   */
+  virtual std::vector<std::string> get_names () const;
+
+  /**
+   * This functions returns information about how the individual components of
```
Comment (jppelteret) [inline include/deal.II/numerics/data_postprocessor.h]: "This function"

File: include/deal.II/numerics/data_postprocessor.h (hunk 2 above)
Comment (bangerth) [reply include/deal.II/numerics/data_postprocessor.h]: Do you know how to do this in doxygen?

Comment (jppelteret) [reply include/deal.II/numerics/data_postprocessor.h]: Maybe with a html unicode character. I think its [`&eacute`](http://www.ascii.cl/htmlcodes.htm)?

**Comment by bangerth** [comment]
I used `Lam&eacute;` and it works indeed. Should be good to go from my perspective!

**Pull request merged**